atlp-rwanda/phoenix-bn-backend
# The user should be able to update his profile

### What does this PR do?
> This PR allow users to update their profile
### Description of Task to be completed
>-update user profile
>-Display user profile
>-upload profile image
>-display profile image
>-write tests
>-write documentations
### How should this be manually tested?
>-Clone the repos
>-Switch to the `ft-updateProfile` branch
>-Add the env variables in `.env` file
>-Your `.env` file should look like this : 
>     - `DB_PORT=`  
>     - `DB_HOST=`  
>     - `DB_USER=`
>     - `DB_PASSWORD=`
>     - `DB_NAME=`  
>     - `DB_DIALECT=`  
>     - `DATABASE_URL=`
>     - `JWTKEY=`
>     - `API_VERSION=v1`
>     - `EMAIL=nomadbarefoot2@gmail.com`
>     - `PASSWORD=nistnvpuhqalrrej`
>     - `CLIENT_ID=978352907060-j1m2j5j0upfajnh6ln5idcnr9f4fk551.apps.googleusercontent.com`
>     - `CLIENT_SECRET=-ABEMIKgnYnvC3RIrhTQLO-F`
>     - `APP_SECRET=5efb78c04b84aead6f3b4587a76d8e2f`
>     - `APP_ID=631208460883265`
>     - `JWT_SECRET_KEY=`
>     - `appPassword=nistnvpuhqalrrej`
>     -`CLOUD_NAME=faustin250`
>     -`CLOUD_API_KEY=886115265422219`
>     -`CLOUD_API_SECRET=3tYIbq4FrNHFj6ix2BWqwc_uvhY`
>     - `HOST=http://localhost:5000`
>     - `PORT=5000`
> run `npm install` to install the required package dependances
> run `npm run db:migrate:undo:all` to undo all migrations
> run `npm start` to run the application
 
### Any background context you want to provide?
>  The user will be able to update his profile details after login to the application
### What are the relevant Trello story?
> Ft-updateProfile >> https://trello.com/c/C1G10LpK/37-update-profile
### Screenshots 
>> login as requester (requester@barefoot.com,pswd:123456)

![1](https://user-images.githubusercontent.com/49835290/98448995-c318ce00-2138-11eb-98f4-e5818e388f1e.PNG)

>Pass auth key into the header and proceed with the following

![4](https://user-images.githubusercontent.com/49835290/98954687-b9131880-2506-11eb-869c-139b497a8db5.PNG)

>Pass auth key into the header and proceed with the following

![3](https://user-images.githubusercontent.com/49835290/98954170-31c5a500-2506-11eb-84c5-d51c892f1e6f.PNG)

>Pass auth key into the header and proceed with the following

![8](https://user-images.githubusercontent.com/49835290/98820765-46873780-2437-11eb-8057-89f3e6f3cff0.PNG)
